src/Test/AppClient.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\Test;
+
+use Http\Adapter\HttpAdapter;
+use Http\Discovery\HttpAdapterDiscovery;
+use Http\Discovery\MessageFactoryDiscovery;
+use Http\Discovery\UriFactoryDiscovery;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Provides a HTTP client for getting responses from your application
+ */
+trait AppClient
```

Comment:
Is this a good name? Perhaps just `HttpClient` or `AppHttpClient` or `HttpAdapter`? I think the name should show that this is *not* a client for your proxy, but a client to simulate requests to your application.